File: uikit/src/main/java/io/agora/chat/uikit/interfaces/OnMenuChangeListener.java
package io.agora.chat.uikit.interfaces;

import android.widget.PopupWindow;

import io.agora.chat.ChatMessage;
import io.agora.chat.uikit.menu.EasePopupWindowHelper;
import io.agora.chat.uikit.menu.MenuItemBean;


public interface OnMenuChangeListener {
    /**
     * Before showing the Menu
     * @param helper
     * @param message
     */
    void onPreMenu(EasePopupWindowHelper helper, ChatMessage message);

    /**
     * Item click
     * @param item
     * @param message
     */
    boolean onMenuItemClick(MenuItemBean item, ChatMessage message);

    /**
     * Dismiss event
     * @param menu
     */
    default void onDismiss(PopupWindow menu) {}
}

File: ease-im-kit/src/main/kotlin/io/agora/chat/uikit/interfaces/OnMenuChangeListener.kt
package io.agora.chat.uikit.interfaces

import io.agora.chat.uikit.common.ChatMessage
import io.agora.chat.uikit.menu.chat.ChatUIKitChatMenuHelper
import io.agora.chat.uikit.model.ChatUIKitMenuItem

interface OnMenuChangeListener {
    /**
     * Before showing the Menu
     * @param helper
     * @param message
     */
    fun onPreMenu(helper: ChatUIKitChatMenuHelper?, message: ChatMessage?)

    /**
     * Item click
     * @param item
     * @param message
     */
    fun onMenuItemClick(item: ChatUIKitMenuItem?, message: ChatMessage?): Boolean

    /**
     * Dismiss event
     */
    fun onDismiss() {}
}